mm2src/coins/eth.rs

```diff
@@ -684,6 +704,14 @@ impl EthCoinImpl {
         let guard = self.erc20_tokens_infos.lock().unwrap();
         (*guard).clone()
     }
+
+    /// # Warning
+    /// Be very careful using this function since it returns dereferenced clone
+    /// of value behind the MutexGuard and makes it non-thread-safe.
+    pub async fn get_nfts_infos(&self) -> HashMap<String, NftInfo> {
+        let guard = self.nfts_infos.lock().await;
+        (*guard).clone()
+    }
```

Comment:
Devs would use this unintentionally without knowing that this is dropping the mutex guard. How about removing function? If we don't care about guard and race-conditions on the value, we can drop the guard in-place rather than providing it as a function.

Reply:
Agree, done